core/core.go

```diff
@@ -475,14 +475,23 @@ func (n *IpfsNode) startOnlineServicesWithHost(ctx context.Context, host p2phost
 
 		var service *pubsub.PubSub
 
+		var pubsubOptions []pubsub.Option
```

Comment:
What happens if they are both set?

Reply:
The user won't be able to send any messages. We should error in that case, shouldn't we...

Reply:
https://github.com/libp2p/go-libp2p-pubsub/pull/122

Reply:
Re-reading this now with some more context it seems like a valid case (although I'm not sure how useful it is), that is, we are requiring signed messages but we don't sign them ourselves.

Reply:
It is a valid case, technically. But it's so useless that we should probably get rid of it.

Reply:
(actually, I may just do that in a followup patch once someone takes a look at that PR)

Reply:
if strict signing is set but signing is disabled, then it will be ignored.

Reply:
Got it. Now that the option validation patch has been merged, it should return an error on construction. That should be the least surprising result.